zrml/swaps/src/lib.rs

```diff
@@ -1384,6 +1386,22 @@ mod pallet {
             Ok(next_pool_id)
         }
 
+        fn destroy_pool(pool_id: PoolId) -> Result<Weight, DispatchError> {
+            let pool = Self::pool_by_id(pool_id)?;
+            let pool_account = Self::pool_account_id(pool_id);
+            for asset in pool.assets.into_iter() {
+                let amount = T::Shares::free_balance(asset, &pool_account);
+                T::Shares::slash(asset, &pool_account, amount);
+            }
+            let pool_share_id = Self::pool_shares_id(pool_id);
+            let (_, liquidity_providers) = T::Shares::accounts_by_currency_id(pool_share_id);
+            T::Shares::destroy_all(pool_share_id, liquidity_providers.iter().cloned());
+            Pools::<T>::remove(pool_id);
+            Self::deposit_event(Event::PoolDestroyed(pool_id));
+            // TODO(#603): Fix weight calculation.
+            Ok(0u64)
```

Comment:
We should not use 0, better use a decent part of the available block weight until we properly calculate it.

Reply:
Okay, I went with 10% of the block. But it doesn't really matter, actually, since we don't use the result of `destroy_pool` in `admin_destroy_market`. I would have to correct the benchmarks for that.

But I don't think it matters, because `admin_destroy_market` currently uses 56% of available block time (worst-case, which assumes that there are 900 accounts that have to be slashed).